workflow-job

```diff
@@ -283,9 +292,38 @@ flows["${sha1}"] = {
                 sh "docker logs ${containers.cassandra}"
             }
             sh "docker run --detach=true --name=${containers.elasticSearch} ${images.elasticSearch}"
-            sh "cd dockerfiles/run/guice; docker build --tag=${images.james} ."
+            sh "cd dockerfiles/run/guice/cassandra; docker build --tag=${images.jamesCassandra} ."
             dockering.success()
-            sh "docker run --detach=true --name=${containers.james} --hostname ${hostname} --expose=143 --publish-all=true ${jamesLinks} ${images.james}"
+            sh "docker run --detach=true --name=${containers.james} --hostname ${hostname} --expose=143 --publish-all=true ${jamesLinks} ${images.jamesCassandra}"
+            
+            echo 'Waiting for James server to be deployed'
+            try {
+                waitForCommandSuccess("docker exec ${containers.james} ${jamesCliWithOptions} listusers", maxRetries)
+            } catch (Exception e) {
+                sh "docker logs ${containers.james}"
+            }
+            deploying.success();
+        }
+        stage "Configuring James Server"
+        node {
+            sh "docker exec ${containers.james} ${jamesCliWithOptions} adddomain domain"
+            sh "docker exec ${containers.james} ${jamesCliWithOptions} adduser imapuser@domain password"
+             
+            echo "Running integration tests on ${buildId}"
+            sh "${findImapPort} && docker run ${verbose} --name=${containers.integration} --entrypoint=\"/root/integration_tests.sh\" ${useMavenCache} --volumes-from=${containers.gitPublish} ${images.jamesCompile} ${integrationArguments}"
+            testing.success();
+        }
+        stage "Deploying Guice Server with JPA ${buildId}"
+        node {
+            sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.keystore}:${keystorePath} destination/conf"
+            sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.jamesCompile}:${jarPathJPA} destination"
+            sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.jamesCompile}:${libPathJPA} destination"
+            sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.jamesCompile}:${cliJarPathJPA} destination"
+            sh "cd dockerfiles/run/guice/jpa; docker cp ${containers.jamesCompile}:${clilibPathJPA} destination"
+            
+            sh "cd dockerfiles/run/guice/jpa; docker build --tag=${images.jamesJPA} ."
+            dockering.success()
+            sh "docker run --detach=true --name=${containers.james} --hostname ${hostname} --expose=143 --publish-all=true ${images.jamesJPA}"
             
             echo 'Waiting for James server to be deployed'
             try {
```

Comment:
This can be extracted in a method (waiting for James server + running tests on it)